InvenTree/plugin/builtin/integration/mixins.py

```diff
@@ -53,6 +58,128 @@ def set_setting(self, key, value, user=None):
+            'repeats': 5,                           # Number of repeats (leave blank for 'forever')
+        }
+    }
+
+    Note: 'schedule' parameter must be one of ['I', 'H', 'D', 'W', 'M', 'Q', 'Y']
+    """
+
+    ALLOWABLE_SCHEDULE_TYPES = ['I', 'H', 'D', 'W', 'M', 'Q', 'Y']
+
+    SCHEDULED_TASKS = {}
+
+    class MixinMeta:
+        MIXIN_NAME = 'Schedule'
+
+    def __init__(self):
+        super().__init__()
+        self.add_mixin('schedule', 'has_scheduled_tasks', __class__)
+        self.scheduled_tasks = getattr(self, 'SCHEDULED_TASKS', {})
+
+        self.validate_scheduled_tasks()
+
+    @property
+    def has_scheduled_tasks(self):
+        return bool(self.scheduled_tasks)
+
+    def validate_scheduled_tasks(self):
+        """
+        Check that the provided scheduled tasks are valid
+        """
+
+        if not self.has_scheduled_tasks:
+            raise ValueError("SCHEDULED_TASKS not defined")
+
+        for key, task in self.scheduled_tasks.items():
+
+            if 'func' not in task:
+                raise ValueError(f"Task '{key}' is missing 'func' parameter")
+
+            if 'schedule' not in task:
+                raise ValueError(f"Task '{key}' is missing 'schedule' parameter")
+
+            schedule = task['schedule'].upper().strip()
+
+            if schedule not in self.ALLOWABLE_SCHEDULE_TYPES:
+                raise ValueError(f"Task '{key}': Schedule '{schedule}' is not a valid option")
+
+            # If 'minutes' is selected, it must be provided!
+            if schedule == 'I' and 'minutes' not in task:
+                raise ValueError(f"Task '{key}' is missing 'minutes' parameter")
+
+    def get_task_name(self, key):
+        # Generate a 'unique' task name
+        slug = self.plugin_slug()
+        return f"plugin.{slug}.{key}"
+
+    def get_task_names(self):
+        # Returns a list of all task names associated with this plugin instance
+        return [self.get_task_name(key) for key in self.scheduled_tasks.keys()]
+
+    def register_tasks(self):
+        """
+        Register the tasks with the database
+        """
+
+        try:
+            from django_q.models import Schedule
+
+            for key, task in self.scheduled_tasks.items():
+
+                task_name = self.get_task_name(key)
+
+                # If a matching scheduled task does not exist, create it!
+                if not Schedule.objects.filter(name=task_name).exists():
+
+                    logger.info(f"Adding scheduled task '{task_name}'")
+
+                    Schedule.objects.create(
+                        name=task_name,
+                        func=task['func'],
+                        schedule_type=task['schedule'],
+                        minutes=task.get('minutes', None),
+                        repeats=task.get('repeats', -1),
+                    )
+        except (ProgrammingError, OperationalError):
+            # Database might not yet be ready
+            pass
```

Comment:
Should we log / retry later? Slow to start up dbs in Kubernetes could cause this to trigger